src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
 
+type Color = 'red' | 'green' | 'blue';
+enum Shapes {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+function roundedArea(area:number):number {
+  return Math.floor(area * 100) / 100;
+}
 
+function chackArg(arg: number):void {
+  if (arg <= 0) {
+    throw new Error('One of side or radius <= 0');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shapes.triangle;
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    [sideA, sideB, sideC].some((side):void => chackArg(side));
 
+    if (sideA + sideB <= sideC
+      || sideB + sideC <= sideA
+      || sideA + sideC <= sideB) {
+      throw new Error('sides 1, 2 and 3 can`t form a triangle');
+    }
+  }
+
+  getArea():number {
+    const semiPerimeter:number = (this.sideA + this.sideB + this.sideC) / 2;
+
+    return roundedArea(Math.sqrt(semiPerimeter
+      * (semiPerimeter - this.sideA)
+      * (semiPerimeter - this.sideB)
+      * (semiPerimeter - this.sideC)));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shapes.circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    chackArg(radius);
+  }
 
+  public getArea():number {
+    return roundedArea(Math.PI * (this.radius ** 2));
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape = Shapes.rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    [width, height].forEach((side):void => chackArg(side));
+  }
+
+  public getArea():number {
+    return roundedArea(this.width * this.height);
+  }
+}
 
+export function getInfo(figure:Rectangle|Circle|Triangle):string {
```

Comment:
You should use here `figure: Figure`, also, please, add everywhere spaces, like that `export function getInfo(figure: Rectangle | Circle | Triangle): string`